packages/@angular/cli/tasks/version-check.ts

```diff
@@ -0,0 +1,75 @@
+const Task = require('../ember-cli/lib/models/task');
+import { CliConfig } from '../models/config';
+import { CliConfig as ConfigInterface } from '../lib/config/schema';
+import { timeToCheck, newVersionAvailable, CheckResult } from '../update/check';
+import { oneLine } from 'common-tags';
+import * as chalk from 'chalk';
+
+export interface VersionCheckTaskOptions {
+  forceCheck: boolean;
+  includeUnstable: boolean;
+}
+
+export default Task.extend({
+  run: function(options: VersionCheckTaskOptions) {
+    const globalConfigManager = CliConfig.fromGlobal();
+    const globalConfig = globalConfigManager.config;
+    const lastChecked = getLastChecked(globalConfig, this.project.root);
+    const frequency = getFrequency(globalConfig);
+
+    const isTimeToCheck = timeToCheck(lastChecked, frequency);
+    if (isTimeToCheck) {
+      updateLastChecked(globalConfigManager, this.project.root);
+    }
+
+    if (options.forceCheck || isTimeToCheck) {
+      const rootDir = '';
+      const packageName = '@angular/cli';
+      return newVersionAvailable(rootDir, packageName, options)
+        .then((result: CheckResult) => {
+          const newVersionAvailable = !!result.newVersion;
+          if (newVersionAvailable) {
+            this.ui.writeLine(chalk.yellow(oneLine`Current version (${result.currentVersion})
```

Comment:
Wording here could be improved with better context and call to action:

```
Your current version of @angular/cli (1.2.3) is out of date. To install the new version, run the following command:
  npm install --save @angular/cli@latest
```

(careful with `--save` vs `--save-dev` depending on where it lives in their package json).